src/main/java/com/piekoszek/nowaksharedrent/auth/AuthController.java

```diff
@@ -1,31 +1,49 @@
 package com.piekoszek.nowaksharedrent.auth;
 
+import com.piekoszek.nowaksharedrent.jwt.JwtData;
+import com.piekoszek.nowaksharedrent.jwt.JwtService;
+import org.mindrot.jbcrypt.BCrypt;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 class AuthController {
 
     private AuthService authService;
+    private JwtService jwtService;
 
-    AuthController (AuthService authService) {
+    AuthController (AuthService authService, JwtService jwtService) {
         this.authService = authService;
+        this.jwtService = jwtService;
     }
 
     @PostMapping("/auth/signup")
-    ResponseEntity<Object> createAccount(@RequestBody Account account) {
+    ResponseEntity<Object> createAccount(@RequestBody Account input) {
```

Comment:
Controller should contain as little logic as possible, so for example it should not calculate hash, so you should pass Account to authService and let it do the job, make it return Optional<JwtData>. Also calculating hash should be extracted to another class or even package... but class should be enough for now.

Reply:
Am I understand correctly that JwtData builder should be moved to AuthService methods?

Reply:
I've tried my best but a bunch of explanations presented on top in google search don't help me that much with Optional.isPresent() along with return usage. IDE suggested some simplification in my `if` statements but I have no idea what can I correct there. Appreciate your help with this ;d

Reply:
input is not the best name, it doesn't say much what it is, accout is much better

Reply:
About IDE suggests, IMO it's more readable like it is now.